### PR TITLE
Dashboard v2 - Uploader

### PR DESCRIPTION
**Pull request by meeh0w**
## ⚠️ Based on #1801, don't merge prematurely :)
* _Reviewing commit-by-commit should be helpful_

## Overview
* Adds some new, shiny icons
* Fixes styles for `Button`
* Improves `Tabs`, so the content fills available space. Also stops unmounting the child component when tab becomes inactive (this will allow the user to change tabs without interrupting any background tasks started by the child component, like uploads).
* Adds `ProgressBar` component
* Copies `Uploader` from `packages/website` and simplifies it a little bit (it contained quite a lot of components we won't need in the dashboard view)
* Adds `Uploader` on the main screen

## 🎥  Visual overview
https://user-images.githubusercontent.com/3853033/156216391-96afd121-44ef-41ae-ac19-20137415991d.mov

### CODE REVIEW

File: packages/dashboard-v2/src/components/Tabs/TabPanel.js [hunk 1]
```diff
@@ -3,17 +3,17 @@ import PropTypes from "prop-types";
...
+  <div
+    role="tabpanel"
+    id={`tabpanel-${tabId}`}
+    aria-labelledby={`tab-${tabId}`}
+    {...props}
+    style={{ display: active ? "block" : "none" }}
+  >
+    {children}
+  </div>
+);
```
Comment (kwypchlo) [inline packages/dashboard-v2/src/components/Tabs/TabPanel.js:16]: I think the component should return null until it is mounted for the first time (active is true for the first time). That way we don't load the invisible tabs and first paint is faster and we potentially save some unnecessary requests if other invisible tabs would request a list of data or sth like that.

File: packages/dashboard-v2/src/components/Uploader/UploaderItem.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
+import * as React from "react";
+import cn from "classnames";
+import bytes from "pretty-bytes";
+import { StatusCodes } from "http-status-codes";
+import copy from "copy-text-to-clipboard";
+import path from "path-browserify";
+import { useTimeoutFn } from "react-use";
+// import ms from "ms";
```
Comment (kwypchlo) [inline packages/dashboard-v2/src/components/Uploader/UploaderItem.js:8]: not needed any more?

**Review by kwypchlo — CHANGES_REQUESTED** [review]
Small comments, otherwise looks good 👍 

Also since you're at it, maybe you can take care of this warning that shows up in github:

Screenshot 2022-03-01 at 18 54 06

File: packages/dashboard-v2/src/components/Tabs/TabPanel.js (hunk 1 above)
Comment (meeh0w) [reply packages/dashboard-v2/src/components/Tabs/TabPanel.js]: Good idea 👍  e3fe7c60

File: packages/dashboard-v2/src/components/Uploader/UploaderItem.js (hunk 2 above)
Comment (meeh0w) [reply packages/dashboard-v2/src/components/Uploader/UploaderItem.js]: Yup. Cleaned it up here 👉  18e198d3

**Review by kwypchlo — APPROVED** [review]

File: packages/dashboard-v2/src/components/Tabs/TabPanel.js [hunk 3]
```diff
@@ -1,15 +1,33 @@
...
 
 /**
  * Besides documented props, it accepts all HMTL attributes a `<div>` element does.
  */
 export const TabPanel = ({ children, active, tabId, ...props }) => {
-  if (!active) {
+  const [wasActivated, setWasActivated] = useState(false);
+
+  useEffect(() => {
+    if (active) {
```
Comment (peterjan) [inline packages/dashboard-v2/src/components/Tabs/TabPanel.js:11]: `(uber) nit` this is redundant, you can just do `setWasActivated(active)`

File: packages/dashboard-v2/src/components/Uploader/buildUploadErrorMessage.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+export default function buildUploadErrorMessage(error) {
+  // The request was made and the server responded with a status code that falls out of the range of 2xx
+  if (error.response) {
+    if (error.response.data.message) {
+      return `Upload failed with error: ${error.response.data.message}`;
+    }
+
+    const statusCode = error.response.status;
+    const statusText = getReasonPhrase(error.response.status);
```
Comment (peterjan) [inline packages/dashboard-v2/src/components/Uploader/buildUploadErrorMessage.js:12]: you can use `statusCode` here

**Review by peterjan — APPROVED** [review]

File: packages/dashboard-v2/src/components/Uploader/buildUploadErrorMessage.js (hunk 4 above)
Comment (meeh0w) [reply packages/dashboard-v2/src/components/Uploader/buildUploadErrorMessage.js:12]: Thanks for this -- I'll address it in future PRs to not cause a chain of changes between this branch and other PRs that depend on it 👌

**Pull request merged**